internal/epoch/epoch_manager.go

```diff
@@ -455,6 +455,28 @@ func (e *Manager) refreshLocked(ctx context.Context) error {
 		}
 	}
 
+	return e.maybeCompactAndCleanup(ctx, *p)
```

Comment:
This is _primarily_ code movement, see below:
- groups all the (async) cleanup tasks in `maybeCompactAndCleanup` and removes them from  `refreshAttemptLocked`
- calls `maybeCompactAndCleanup` here in `refreshLocked`, which happens _after_ `refreshAttemptLocked` has completed.

This does NOT disable index maintenance on the read path yet.
This (probably) reduces the number of background tasks that get launched in the case of failed index loads, that is multiple calls to `refreshAttemptLocked` due to retries.

Which in turn, _may_ reduce the probability of retries that may have been caused by the interference (additional resource requirements) introduced by these background tasks.